### ISSUE TITLE
Add Docker 24.0.x version

### ISSUE THREAD

**Issue by rayandas** [rancher/rancher#41851]
https://docs.docker.com/engine/release-notes/24.0/

Following docker versions to be added:
```
24.0.0
24.0.1
24.0.2
```
Same approach as https://github.com/rancher/rancher/issues/40731 to be followed for implementation and testing.

SURE-6491

**Comment by antoinetran** [comment]
Hi, I have issue to find which version of https://github.com/rancher/rancher/releases or rke has the fix for docker-ce 23. For docker-ce 24, **is-it possible that this information is marked somewhere** (either here or in release note)? Our project use rancher and we are waiting for docker-ce-23 or 24 to be fixed and released but nowhere can we find in which version it is fixed. Thank you!

**Comment by snasovich** [comment]
@antoinetran , Docker 23 support will be included with the next (2.7.5) release per https://github.com/rancher/rancher/issues/40731. It should be included on the release notes as well as on support matrix published after the release (e.g. next to current one for 2.7.4 - https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-4/).

Docker 24 support is planned for the later release (Q3 one per the milestone set - not a commitment though) as per this issue.

**Comment by rishabhmsra** [comment]
Validated below docker versions on different OSes by creating node driver and custom clusters :
- https://releases.rancher.com/install-docker-dev/24.0.0.sh
- https://releases.rancher.com/install-docker-dev/24.0.1.sh
- https://releases.rancher.com/install-docker-dev/24.0.2.sh

- **Note :** On `CentOS 8 stream` docker installation script is failing with below error. So tested the cluster provisioning using the docker upstream script `https://get.docker.com` and the provisioning was successful.
```
Errors during downloading metadata for repository 'appstream':
  - Status code: 404 for https://vault.centos.org/centos/8-stream/AppStream/x86_64/os/repodata/repomd.xml (IP: <REDACTED>)
Error: Failed to download metadata for repo 'appstream': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```
- Script is working on CentOS 8 (non-stream version), but that is EOL.

**Comment by rishabhmsra** [comment]
OS to test | Docker Version | Custom Cluster | Node Driver | Notes
-- | -- | -- | -- | --
Ubuntu 22.04 | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
Ubuntu 20.04 | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
Ubuntu 18.04 | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
Oracle Linux  | 24.0.0/24.0.1/24.0.2 | PASS | Fail | Fails due to this issue - https://github.com/rancher/rancher/issues/38214
Oracle Linux  | 24.0.0/24.0.1/24.0.2 | PASS | Fail | Fails due to this issue - https://github.com/rancher/rancher/issues/38214
Oracle Linux 7  | 24.0.0/24.0.1/24.0.2 | PASS | Fail | Fails due to this issue - https://github.com/rancher/rancher/issues/38214
CentOS 7.9 | 24.0.0/24.0.1/24.0.2 | PASS | PASS  |  
RHEL  | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
RHEL  | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
Rocky  | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
Rocky  | 24.0.0/24.0.1/24.0.2 | PASS | PASS |  
SLES:12SP5 | 24.0.0/24.0.1/24.0.2 |   |   | Comes with docker pre-installed
SLES:15SP4 | 24.0.0/24.0.1/24.0.2 |   |  | Comes with docker pre-installed
SLES:15SP5 | 24.0.0/24.0.1/24.0.2 |   |  | Comes with docker pre-installed
OpenSUSE:15.4 | 24.0.0/24.0.1/24.0.2 |   |  | Comes with docker pre-installed
 OpenSUSE:15.5 | 24.0.0/24.0.1/24.0.2 |   |  | Comes with docker pre-installed

**Comment by rishabhmsra** [comment]
Docker install scripts are working as expected hence moving it to release-task, @kinarashah can you please let us know if there's anything else needs to be done before we mark the ticket as done.

**Comment by kinarashah** [comment]
The new docker versions have been released, you can mark the issue as done if there are no additional checks required after the release step. 

Following were released:
https://releases.rancher.com/install-docker/24.0.0.sh
https://releases.rancher.com/install-docker/24.0.1.sh
https://releases.rancher.com/install-docker/24.0.2.sh

And https://releases.rancher.com/install-docker/24.0.sh points to latest 24.0.2.
